actionpack/lib/action_controller/metal/request_forgery_protection.rb

```diff
@@ -153,11 +157,31 @@ module ClassMethods
       #    class ApplicationController < ActionController:x:Base
       #      protect_from_forgery with: CustomStrategy
       #    end
+      # * <tt>:store</tt> - Set the strategy to store and retrieve CSRF tokens.
+      #
+      # Built-in session token strategies are:
+      # * <tt>:session</tt> - Store the CSRF token in the session.  Used as default if <tt>:store</tt> option is not specified.
+      # * <tt>:cookie</tt> - Store the CSRF token in an encrypted cookie.
+      #
+      # You can also implement custom strategy classes for CSRF token storage:
+      #
+      #   class CustomStore < ActionController::RequestForgeryProtection::AbstractStore
+      #     def retrieve_token(request)
+      #       custom_location ||= generate_csrf_token
+      #     end
+      #
+      #    class ApplicationController < ActionController:x:Base
+      #      protect_from_forgery store: CustomStore
+      #    end
+      #   end
       def protect_from_forgery(options = {})
         options = options.reverse_merge(prepend: false)
 
         self.forgery_protection_strategy = protection_method_class(options[:with] || :null_session)
         self.request_forgery_protection_token ||= :authenticity_token
+
+        self.csrf_token_storage_strategy = options[:store] || SessionStore.new
```

Comment:
It may makes sense to accept `symbols` like `protection_method_class` does. But no big deal.

Reply:
Yeah... I struggled with that one... if we pass in a symbol, then we need to use the default cookie name (which is probably ok).  I'll add the symbols back.

Reply:
Yeah, you can use accept instances, maybe just check if they `respond_to?(...)`